src/frontend/src/lib/components/carousel/Carousel.svelte

```diff
@@ -0,0 +1,235 @@
+<script lang="ts">
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { onMount } from 'svelte';
+	import Controls from '$lib/components/carousel/Controls.svelte';
+	import Indicators from '$lib/components/carousel/Indicators.svelte';
+	import {
+		updateSliderFrameVisiblePart,
+		buildCarouselSliderFrame,
+		disableTransition,
+		enableTransition
+	} from '$lib/utils/carousel.utils';
+
+	export let autoplay = 5000;
+	export let duration = 300;
+	export let easing = 'ease-out';
+	export let styleClass: string | undefined = undefined;
+
+	/**
+	 * Carousel container element variables
+	 */
+	let container: HTMLDivElement | undefined;
+	let containerWidth = 0;
+
+	/**
+	 * Computed slider frame element that wraps all slides
+	 */
+	let sliderFrame: HTMLDivElement | undefined;
+
+	/**
+	 * Variables related to the slides
+	 */
+	let slides: Node[];
+	let currentSlide = 0;
+	let totalSlides: number;
+
+	/**
+	 * Autoplay timer
+	 */
+	let autoplayTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * Slide transition timer that needed for last-to-first and first-to-last non-animated transform
+	 */
+	let slideTransformTimer: NodeJS.Timeout | undefined = undefined;
+
+	/**
+	 * A const to be used for enabling transition animation
+	 */
+	const animationParams = {
+		easing,
+		duration
+	};
+
+	/**
+	 * Initialise all required data and attach/detach events
+	 */
+	onMount(() => {
+		initializeCarousel();
```

Comment:
The carousel is not dynamic and can only be initialized once, is this expected?

Reply:
Yes, it only needs to be initialised on mount and on resize.

Reply:
Can you add a comment somewhere because this is kind of a limitation.

Reply:
Adding now, missed this comment.